opencensus/common/configuration/__init__.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+
+__all__ = ['Namespace', 'eval']
+
+
+class Namespace(object):
```

Comment:
@c24t, need your help to pick a better name.

Reply:
This is a pretty amazing little class, but I think I'm missing something here. Do you want to use `Namespace` objects to do some complicated config? Or do you expect only to use it via `load`?

As I understand it, this class lets you `eval` without having to import `opencensus` explicitly. It's a callable that (recursively creates callables that) lazily imports all the ancestor modules of the named target.

How is using a `Namespace` here different than using the imported module directly? E.g.

```python
import opencensus

def load(expr):
    return eval(expr, {}, {'opencensus': opencensus})
```

Reply:
Import `opencensus` won't work in several conditions:

```
Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 23:09:28) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import opencensus
>>> opencensus.ext.azure.trace_exporter.AzureExporter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'opencensus' has no attribute 'ext'
>>> import opencensus.ext.azure.trace_exporter
>>> opencensus.ext.azure.trace_exporter.AzureExporter
<class 'opencensus.ext.azure.trace_exporter.AzureExporter'>
>>>
```

Reply:
That makes sense, turns out my example only worked because I'd already imported all the parent modules of my test class by calling your `load` first.

Reply:
What about `Loader` or `AutoImporter`? Not obviously better than `Namespace`, this class my be too weird to name well.

Reply:
I will keep Namespace for now, as it is weird to have `Loader.parent` being another `Loader` (same applied for `AutoImporter`). Would be open for a better name.